ListApp/Tests/ListApp.Api.Tests/ItemControllerTests.cs

```diff
@@ -62,60 +94,69 @@ public void SetUp()
             Assert.AreEqual(expectedResponseCode, responseMessage.StatusCode);
             Assert.AreEqual(expectedLocation, responseMessage.Headers.Location.ToString());
             Assert.IsTrue(responseMessage.TryGetContentValue(out ListItem responseItem));
             Assert.That(responseItem, Is.EqualTo(expectedItem).UsingListItemComparer());
         }
 
         [Test]
-        public async Task Put_ValidItem_ResponseIsOfCorrectTypeAndReturnsDefaultItemWithCorrectLocation()
+        public async Task Put_ValidItem_ResponseIsOfCorrectTypeAndReturnsDefaultItemWithCorrectLocationAndCallsRepoAddAndDeleteOnce()
         {
-            var expectedLocation = _itemsController.Url.Request.RequestUri + $"/{PostedItemGuid}";
-            const HttpStatusCode expectedResponseCode = HttpStatusCode.Created;
+            const HttpStatusCode expectedResponseCode = HttpStatusCode.OK;
             var expectedItem = new ListItem
             {
-                Id = PostedItemGuid,
-                Text = "Create another ListItem item!"
+                Id = Guid.Empty,
+                Text = "Stretch correctly"
             };
 
-            var receivedResponse = await _itemsController.PutAsync(Guid.NewGuid(), PostedItem);
+            var receivedResponse = await _itemsController.PutAsync(PostedItemGuid, PostedItem);
             var responseMessage = await receivedResponse.ExecuteAsync(CancellationToken.None);
 
+            Assert.DoesNotThrowAsync(() => _itemsRepository.Received(1).UpdateAsync(PostedItemGuid, PostedItem));
             Assert.AreEqual(expectedResponseCode, responseMessage.StatusCode);
-            Assert.AreEqual(expectedLocation, responseMessage.Headers.Location.ToString());
             Assert.IsTrue(responseMessage.TryGetContentValue(out ListItem responseItem));
             Assert.That(responseItem, Is.EqualTo(expectedItem).UsingListItemComparer());
         }
 
         [Test]
-        public async Task Delete_WithAnyId_ReturnsNoContent()
+        public async Task Delete_WithAnyId_ResponseIsOfCorrectTypeAndReturnsDefaultItemAndCallsRepoDeleteAsyncOnce()
         {
-            const HttpStatusCode expectedResponseCode = HttpStatusCode.NoContent;
+            const HttpStatusCode expectedResponseCode = HttpStatusCode.OK;
+            var expectedItem = new ListItem
+            {
+                Id = Guid.Empty,
+                Text = "Stretch correctly"
+            };
 
-            var receivedResponse = await _itemsController.DeleteAsync(Guid.NewGuid());
+            var receivedResponse = await _itemsController.DeleteAsync(PostedItemGuid);
             var responseMessage = await receivedResponse.ExecuteAsync(CancellationToken.None);
 
+            Assert.DoesNotThrowAsync(() => _itemsRepository.Received(1).DeleteAsync(PostedItemGuid));
             Assert.AreEqual(expectedResponseCode, responseMessage.StatusCode);
+            Assert.IsTrue(responseMessage.TryGetContentValue(out ListItem responseItem));
```

Comment:
You don't need to be this defensive in the tests. Just get the content value in the Act part of the test and then check it in the last Assert.